### PR TITLE
Supercruise - Optimised + Improved collision detection

### PR DESCRIPTION
**Pull request by PowerfulBacon**
Will need to update for https://github.com/BeeStation/BeeStation-Hornet/pull/5278 when thats merged.

~~DO NOT MERGE UNTIL https://github.com/BeeStation/BeeStation-Hornet/pull/5278 OTHERWISE 5278 WILL BE BROKEN.~~ Ready

## About The Pull Request

This PR aims to improve the optimisation of supercruise maps for large numbers of objects, as well as making it so that if you pass through an object you will collide with it.

- [x] Optimise orbital objects into collision zones. Only check collisions with zones that are being intersected with.
- [x] Don't bother checking collisions with objects we don't have collision effects with.
- [x] Make it so passing through an object will trigger a collision event. (Either the 2 circles need to touch, or the 2 vectors need to intersect)

How?:
image

image

Does some pretty simple vector math to see if 2 velocities are intersecting.
mu = (vx * py2 + vy * px - py * vx - vy * px2) / (vy * vx2 - vx * vy2)
lambda = (px2 + vx2 * mu - px) / vx
if mu and lambda are less than 1 and greater than 0 they are intersecting.

## Why It's Good For The Game

Will improve supercruise collision detection and allow for larger amounts of objects on the map without causing O(n^2) levels of lag. ~~Useful if I ever want to add projectiles~~

## Changelog
:cl:
code: Improves the supercruise collision detection by optimising it and making it so passing something triggers a collision event.
code: SSorbits now supports multiple orbital maps.
/:cl:

### CODE REVIEW

File: code/modules/shuttle/super_cruise/orbital_map_components/orbital_object.dm [hunk 1]
```diff
@@ -142,6 +225,46 @@
...
+			if((vx || vy) && (vx2 || vy2))
+				//Collision between 2 vectors using simultaneous equations.
+				var/mu = (vx * py2 + vy * px - py * vx - vy * px2) / (vy * vx2 - vx * vy2)
+				var/lambda = (px2 + vx2 * mu - px) / vx
+				if(lambda >= 0 && lambda <= 1 && mu >= 0 && mu <= 1)
+					//Collision
+					LAZYADD(colliding_with, object)
+					collision(object)
+					colliding = TRUE
+					message_admins("2 objects just intersected and thats really cool!")
```
Comment (PowerfulBacon) [inline code/modules/shuttle/super_cruise/orbital_map_components/orbital_object.dm]:
```suggestion
```

**Comment by ike709** [comment]
based

**Comment by francinum** [comment]
qwer2d2

**Comment by ike709** [comment]
This good to go?

**Comment by PowerfulBacon** [comment]
Yea

**Pull request merged**